source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -737,13 +847,38 @@ bool TrySetKubectl()
 
                 if (TryExecuteKubectlCommand("version", "--client", "--short"))
                 {
+                    TrySetKubectlVersion();
                     return true;
                 }
 
                 log.Error($"Could not find kubectl. Make sure {kubectl} is on the PATH. See https://g.octopushq.com/KubernetesTarget for more information.");
                 return false;
             }
 
+            void TrySetKubectlVersion()
+            {
+                var kubectlVersionOutput = ExecuteCommandAndReturnOutput(kubectl, "version", "--client", "--output=json");
+                var kubeCtlVersionJson = string.Join(" ", kubectlVersionOutput);
+                string kubectlVersionString;
+                try
+                {
+                    var clientVersion = JsonConvert.DeserializeAnonymousType(kubeCtlVersionJson, new { clientVersion = new { gitVersion = "1.0.0" } });
+                    kubectlVersionString = clientVersion?.clientVersion?.gitVersion?.TrimStart('v');
+                    if (kubectlVersionString != null)
+                    {
+                        kubectlVersion = Maybe<Version>.Some(new Version(kubectlVersionString));
+                        return;
+                    }
+                }
+                catch (Exception)
+                {
```

Comment:
This is a bit of a code smell/redundant is it worth changing?

Reply:
I am not too sure about the context here, but here I could not easily understand what kind of Exception we are trying to handle here. Reading the code inside the `try` block, I think the Exception might come from JSON serialisation. In this case I think it would be better if we preserve the original Exception, or at least log it out, rather than swallowing it. It may help future debugging.
For example, in the future we could realise that the version isn't set, which caused some bugs in the code, but I imagine it would take a lot of debugging or reasoning about the code logic to find out it is the JSON format that caused the error

Reply:
I was just trying to ignore any possible exceptions from the kubectl version command. I've tested this command on older versions of kubectl but just have some concerns hence the try block. This version is only used for as a part of fetching the EKS api version and I think the Maybe<> handles the case of it not being set but I'll add logging in case it's not.

Reply:
I've searched for the usages of `kubeCtlVersion`, and it was only read from 1 place. I think it will perhaps be clearer if we make this a return value of this method, and include this as a parameter in methods that needs to access it.

Thereby we could avoid the style of invoking a method for the side effect which changes a global state (i.e. kubeCtlVersion). This also make this part of logic more testable, and give us more confidence that the `catch` block only hides problems associated with version parsing